### PR TITLE
Issue #KN-316 feat: Pdf player: web component implementation for knowlg portal

### PR DESCRIPTION
**Pull request by santoshilimi**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -78,6 +77,7 @@ import { PublishedPopupComponent } from './players/published-popup/published-pop
     MatListModule,
     MatPaginatorModule
   ],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (HarishGangula) [inline src/app/app.module.ts:80]: WHy the custom element scema is required ?

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts (hunk 1 above)
Comment (santoshilimi) [reply src/app/app.module.ts:80]: Angular by default will not recognise our custom sunbird-pdf-player HTML tag, to help Angular know our Web Component we need to add a schemas property to the @NgModule decorator and add CUSTOM_ELEMENTS_SCHEMA inside the schemas array.

**Review by HarishGangula — APPROVED** [review]

File: src/app/players/base-player/base-player.component.ts [hunk 2]
```diff
@@ -52,10 +52,10 @@ export class BasePlayerComponent implements OnDestroy {
   }
 
   playerEvents(event) {
-    if (event?.edata?.type === 'SHARE') {
+    if (event?.detail?.edata?.type === 'SHARE') {
```
Comment (vinukumar-vs) [inline src/app/players/base-player/base-player.component.ts]: Use always global constants for any strings.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: src/app/players/base-player/base-player.component.ts (hunk 2 above)
Comment (santoshilimi) [reply src/app/players/base-player/base-player.component.ts]: Done

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**